enclave-runtime/src/tls_ra/seal_handler.rs

```diff
@@ -105,6 +109,22 @@ where
 		info!("Successfully updated shard {:?} with provisioned state", shard);
 		Ok(())
 	}
+
+	/// Seal an empty, newly initialized state.
+	///
+	/// Requires the shielding key to be sealed and updated before calling this.
+	///
+	/// Call this function in case we don't provision the state itself, only the shielding key.
+	/// Since the enclave signing account is derived from the shielding key, we need to
+	/// newly initialize the state with the updated shielding key.
+	fn seal_new_empty_state(&self, shard: &ShardIdentifier) -> EnclaveResult<()> {
+		let enclave_account =
+			self.shielding_key_repository.retrieve_key()?.derive_ed25519()?.public();
+		let state = Stf::init_state(enclave_account.into());
+		self.state_handler.reset(state, shard)?;
+		info!("Successfully reset state with new enclave account, for shard {:?}", shard);
+		Ok(())
+	}
```

Comment:
Makes sense, at first, I thought this is a hack, but I think it isn't. It will always need to initialize the state afterwards. However, I think all the non-primary workers initialize a state in the beginning, even though they don't need to. We could change this default behaviour.

Reply:
We can re-think the way we initialize state - I'm sure there is room for improvement. At the moment initialization is always required, because the state gets initialized many steps before we know if we're the primary or secondary validateer.

Reply:
Yes, I agree, and as we know fundamental changes to a bootstrapping flow are always tedious XD